### PR TITLE
Update valve configuration and control cluster XML according to latest spec

### LINKED ISSUE #30897: [1.3] Valve and Boolean Sensor renamed in latest spec
The valve device type has been renamed to Water Valve in the latest spec, we need to align the SDK with this latest change.

The Boolean Sensor Configuration cluster has been renamed to Boolean State Configuration, we need to align the SDK with this latest change.

Spec PR's
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8649
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8648

### PR DESCRIPTION
**Pull request by fessehaeve**
keeps XML up to date with spec fixes #30897

### CODE REVIEW

File: src/controller/data_model/controller-clusters.matter [hunk 1]
```diff
@@ -4063,25 +4068,29 @@ provisional cluster ValveConfigurationAndControl = 129 {
...
   }
 
-  attribute access(write: manage) nullable elapsed_s openDuration = 0;
-  readonly attribute optional nullable epoch_us autoCloseTime = 1;
-  readonly attribute optional nullable elapsed_s remainingDuration = 2;
-  readonly attribute nullable ValveStateEnum currentState = 3;
-  readonly attribute nullable ValveStateEnum targetState = 4;
-  attribute access(write: manage) optional ValveStateEnum startUpState = 5;
+  readonly attribute nullable elapsed_s openDuration = 0;
+  readonly attribute nullable elapsed_s defaultOpenDuration = 1;
```
Comment (andy31415) [inline src/controller/data_model/controller-clusters.matter:4086]: DefaultOpenDuration seems to be r/w not read-only in the spec

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml [hunk 2]
```diff
@@ -48,51 +57,37 @@ limitations under the License.
...
-    <attribute side="server" code="0x0001" define="AUTO_CLOSE_TIME" type="epoch_us" isNullable="true" writable="false" optional="true">AutoCloseTime</attribute>
-    <attribute side="server" code="0x0002" define="REMAINING_DURATION" type="elapsed_s" isNullable="true" writable="false" optional="true">RemainingDuration</attribute>
-    <attribute side="server" code="0x0003" define="CURRENT_STATE" type="ValveStateEnum" isNullable="true" writable="false" optional="false">CurrentState</attribute>
-    <attribute side="server" code="0x0004" define="TARGET_STATE" type="ValveStateEnum" isNullable="true" writable="false" optional="false">TargetState</attribute>
-    <attribute side="server" code="0x0005" define="START_UP_STATE" type="ValveStateEnum" isNullable="false" writable="true" optional="true">
-      <description>StartUpState</description>
-      <access op="read" privilege="view"/>
-      <access op="write" privilege="manage"/>
-    </attribute>
+    <attribute side="server" code="0x0000" define="OPEN_DURATION" type="elapsed_s" isNullable="true" writable="false" optional="false">OpenDuration</attribute>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:60]: Are we missing the min 1 constraint on OpenDuration and DefaultOpenDuration?

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml [hunk 3]
```diff
@@ -48,51 +57,37 @@ limitations under the License.
...
-    </attribute>
+    <attribute side="server" code="0x0008" define="DEFAULT_OPEN_LEVEL" type="percent" isNullable="false" min="1" max="100" writable="true" default="100" optional="true">DefaultOpenLevel</attribute>
     <attribute side="server" code="0x0009" define="VALVE_FAULT" type="ValveFaultBitmap" isNullable="false" writable="false" optional="true">ValveFault</attribute>
 
     <command source="client" code="0x00" name="Open" optional="false">
-      <description>This command is used to set the valve to its fully open position.</description>
-      <arg name="OpenDuration" type="elapsed_s" optional="true"/>
+      <description>This command is used to set the valve to its open position.</description>
+      <arg name="OpenDuration" type="elapsed_s" optional="true" isNullable="true"/>
+      <arg name="TargetLevel" type="percent" optional="true"/>
```
Comment (ReneJosefsen) [inline src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:76]: Not sure if we have this on these fields, but do we need the min 1 constraint?

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml (hunk 2 above)
Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:60]: Look at the comment above this line. There is an issue tracking this.

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml (hunk 3 above)
Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:76]: If you mean min and max, no we don't have it for command fields.

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml (hunk 2 above)
Comment (ReneJosefsen) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:60]: Arh ok, I see. Had to re-read the comment before it made sense. 👍🏻

**Review by ReneJosefsen — APPROVED** [review]

**Review by cecille — APPROVED** [review]
Verified against a new scrape of the spec, have the following diff:
< cluster ValveConfigurationAndControl = 129 {
---
> provisional cluster ValveConfigurationAndControl = 129 {
5a6,9
>   enum StatusCodeEnum : enum8 {
>     kFailureDueToFault = 2;
>   }
> 
17c21
<   bitmap ValveFaultBitmap : bitmap8 {
---
>   bitmap ValveFaultBitmap : bitmap16 {

Here, all the differences are on the scraper side - the scrape of the ValveFaultBitmap is incorrect - bitmap16 is correct here and in the spec, scrape doesn't include the status or provisional. TL;DR, new XML appears to match spec, it's good.

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml [hunk 4]
```diff
@@ -19,21 +19,30 @@ limitations under the License.
...
-    <item name="Open" value="0x0"/>
-    <item name="Closed" value="0x1"/>
+    <item name="Closed" value="0x0"/>
+    <item name="Open" value="0x1"/>
+    <item name="Transitioning" value="0x2"/>
+  </enum>
+
+  <enum name="StatusCodeEnum" type="enum8">
+    <cluster code="0x0081"/>
+     <item name="FailureDueToFault" value="0x02"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: Spec seems to have FAILURE_DUE_TO_FAULT?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml (hunk 4 above)
Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: Do you mean this misses the underscores? I thought this is the format for enums and such types?

Comment (fessehaeve) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: I'm resolving this but we can continue the discussion.

Comment (ReneJosefsen) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: Let me try to check other clusters. This is a bit different from cluster to cluster 🫤

Comment (ReneJosefsen) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: From what I can see, most other cluster uses PascalCase for the status code names, where Disco Ball and IM uses the UpperCase with "_". 

With that said, I could also see that the IM status codes are "translated" into PascalCase as well, e.g. CONSTRAINT_ERROR becomes ConstraintError. So not really sure what is the best path forward here. @bzbarsky-apple any suggestions?

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: @ReneJosefsen @fessehaeve We are going to need to generate the XML from the spec.  They should match.  In this case, I suggest a spec issue to fix the spec.

Comment (ReneJosefsen) [reply src/app/zap-templates/zcl/data-model/chip/valve-configuration-and-control-cluster.xml:45]: Not a problem, I can easily adjust this on the spec side. I will get a PR up for this asap.

**Pull request merged**